sklego/meta.py

```diff
@@ -100,7 +152,69 @@ def __remove_groups_from_x(self, X):
         else:
             return np.delete(X, self.groups, axis=1)
 
-    def fit(self, X, y):
+    def __fit_grouped_estimator(self, X, y, value_columns, group_columns):
+        group_indices = X.groupby(group_columns).indices
+
+        try:
+            grouped_estimations = {
+                group: clone(self.estimator).fit(X.loc[indices, value_columns], y.loc[indices])
+                for group, indices in group_indices.items()
+            }
+        except KeyError:
+            raise KeyError("The indices of y and X should match")
+
+        return grouped_estimations
+
+    def __check_shrinkage_func(self):
+        """Validate the shrinkage function"""
+        group_lengths = [10, 5, 2]
+        expected_shape = np.array(group_lengths).shape
```

Comment:
I am in doubt if this check makes sense here. This is basically a test. 

The default shrinkage functions should be tested in the tests. And it might be fair to expect the user to supply a valid shrinkage function themselves.

Reply:
Or only check the function when it is given by the user.

Reply:
I chose the latter option